distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso

```diff
@@ -122,24 +122,31 @@ type Column
        - operands: A vector of additional operation arguments (the column itself
          is always passed as the first argument).
        - new_name: The name of the resulting column.
-    make_op self op_kind operands new_name =
+       - include_self: If True, then self is used as the first argument;
+         otherwise, it is omitted.
+    make_op : Text -> Vector Text -> (Text | Nothing) -> Boolean -> Column
+    make_op self op_kind operands new_name include_self=True =
         checked_support = if self.connection.dialect.is_supported op_kind then True else
             Error.throw (Unsupported_Database_Operation.Error "The operation "+op_kind+" is not supported by this backend.")
         checked_support.if_not_error <|
             type_mapping = self.connection.dialect.get_type_mapping
             prepare_operand operand = case operand of
                 other_column : Column ->
-                    if Helpers.check_integrity self other_column then other_column.expression else
+                    ## self and other_column do not have to be compatible if self is not going to be used as an argument to the op.
+                    if include_self == False || Helpers.check_integrity self other_column then other_column.expression else
```

Comment:
I do not agree. The column is still created _in the context_ of the `self`-column (that's why your const helper is a member method and not a static - because it needs to be in the same context). So the context-compatibility must be checked regardless.

Reply:
Done.